modules/renter/download.go

```diff
@@ -317,6 +321,7 @@ func (cd *chunkDownload) recoverChunk() error {
 		// Signal that the download is complete.
 		cd.download.downloadComplete = true
 		close(cd.download.downloadFinished)
+		return cd.download.destination.Close()
```

Comment:
This is not a good code pattern, because you've got a return statement in a nested situation. If at some point in the future we were to modify this function to add things at the end, we may not realize that it's not running in this block of code. So here we'd definitely want the full `if err != nil` block, instead of just returning whatever value.